csharp/Microsoft.Azure.Databricks.Client/PipelinesApiClient.cs

```diff
@@ -0,0 +1,223 @@
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Databricks.Client;
+
+public class PipelinesApiClient : ApiClient, IPipelinesApi
+{
+    public PipelinesApiClient(HttpClient httpClient) : base(httpClient)
+    {
+    }
+
+    public async Task<PipelinesList> List(int maxResults = 25, string pageToken = default, CancellationToken cancellationToken = default)
+    {
+        if (maxResults < 1 || maxResults > 100)
+        {
+            throw new ArgumentOutOfRangeException(nameof(maxResults), "limit must be between 1 and 100");
+        }
+
+        StringBuilder requestUriSb = new($"{ApiVersion}/pipelines?max_results={maxResults}");
+
+        if (!string.IsNullOrEmpty(pageToken))
+        {
+            requestUriSb.Append($"&pageToken={pageToken}");
+        }
+
+        var requestUri = requestUriSb.ToString();
+        return await HttpGet<PipelinesList>(this.HttpClient, requestUri, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<(string, PipelineSpecification)> Create(
+        PipelineSpecification pipelineSpecification,
+        bool dryRun = true,
+        bool allowDuplicateNames = false,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines";
+        var request = JsonSerializer.SerializeToNode(pipelineSpecification, Options)!.AsObject();
+        request.Add("dry_run", dryRun);
+        request.Add("allow_duplicate_names", allowDuplicateNames);
+
+        var response = await HttpPost<JsonObject, JsonObject>
+                (this.HttpClient, requestUri, request, cancellationToken)
+            .ConfigureAwait(false);
+
+        string pipelineId = null;
+        if (response["pipeline_id"] != null)
+        {
+            pipelineId = response["pipeline_id"].GetValue<string>();
+        }
+
+        PipelineSpecification effectiveSettings = null;
+        if (response["effective_settings"] != null)
+        {
+            effectiveSettings = JsonSerializer.Deserialize<PipelineSpecification>(response["effective_settings"].ToJsonString(), Options);
+        }
+
+        return (pipelineId, effectiveSettings);
+    }
+
+    public async Task Edit(
+        string pipelineId,
+        PipelineSpecification pipelineSpecification,
+        bool allowDuplicateNames = false,
+        int? expectedLastModified = null,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}";
+        var request = JsonSerializer.SerializeToNode(pipelineSpecification, Options)!.AsObject();
+        request.Add("pipeline_id", pipelineId);
+        request.Add("allow_duplicate_names", allowDuplicateNames);
+
+        if (expectedLastModified != null)
+        {
+            request.Add("expected_last_modified", expectedLastModified.Value);

```

Comment:
I believe you can simply pass in a DateTimeOffset object into the JsonObject, and the JsonConverter should be able to convert it into a `long`.